src/plugins/shared_dictionaries_plugin/shared_dictionaries_column_processor.hpp

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#include "hyrise.hpp"
+#include "shared_dictionaries_plugin.hpp"
+#include "storage/table.hpp"
+#include "utils/size_estimation_utils.hpp"
+
+namespace opossum {
+
+/**
+ * Processes one column for the SharedDictionariesPlugin
+ */
+template <typename T>
+class SharedDictionariesColumnProcessor {
+  friend class SharedDictionariesPluginTest;
+
+ public:
+  const std::shared_ptr<Table> table;
+  const std::string& table_name;
+  const ColumnID column_id;
+  const std::string& column_name;
+  const double jaccard_index_threshold;
+  std::shared_ptr<SharedDictionariesPlugin::SharedDictionariesStats> stats;
+  SharedDictionariesColumnProcessor(
+      const std::shared_ptr<Table>& init_table, const std::string& init_table_name, const ColumnID init_column_id,
+      const std::string& init_column_name, const double init_jaccard_index_threshold,
+      const std::shared_ptr<SharedDictionariesPlugin::SharedDictionariesStats>& init_stats);
+
+  void process();
+
+  /**
+   * Merge plan for a shared dictionary containing the segments to merge and additional information
+   */
+  struct MergePlan {
+   public:
+    std::shared_ptr<const pmr_vector<T>> shared_dictionary;
+    std::vector<SegmentChunkPair<T>> segment_chunk_pairs_to_merge;
+    bool contains_non_merged_segment;
+    bool contains_already_merged_segment;
+    uint64_t non_merged_total_bytes = 0ul;
+    uint64_t non_merged_dictionary_bytes = 0ul;
```

Comment:
Specifying `ul` when the data type is clear seems redundant.

```suggestion
    uint64_t non_merged_total_bytes = 0;
    uint64_t non_merged_dictionary_bytes = 0;
```

Reply:
Fixed in https://github.com/hyrise/hyrise/pull/2394/commits/7f88317d0867f9bfd92688f5061daea73a39be35